generator/src/lib.rs

```diff
@@ -1362,22 +1360,21 @@ pub fn generate_enum<'a>(
 }
 
 pub fn generate_result(ident: Ident, enum_: &vk_parse::Enums) -> TokenStream {
-    let notation = enum_.children.iter().filter_map(|elem| {
-        let (variant_name, notation) = match *elem {
-            vk_parse::EnumsChild::Enum(ref constant) => (
-                constant.name.as_str(),
-                constant.comment.as_deref().unwrap_or(""),
-            ),
-            _ => {
-                return None;
-            }
-        };
-
-        let variant_ident = variant_ident(enum_.name.as_ref().unwrap(), variant_name);
-        Some(quote! {
-            Self::#variant_ident => Some(#notation)
+    let notation = enum_
+        .children
+        .iter()
+        .filter_map(|elem| match *elem {
+            vk_parse::EnumsChild::Enum(ref constant) => Some(constant),
+            _ => None,
         })
-    });
+        .map(|constant| {
+            let variant_name = constant.name.as_str();
+            let notation = constant.comment.as_deref().unwrap_or("");
+            let variant_ident = variant_ident(enum_.name.as_ref().unwrap(), variant_name);
+            quote! {
+                Self::#variant_ident => Some(#notation)
+            }
+        });
```

Comment:
I'm not sure this is an improvement; we're replacing a single `filter_map` with a chained `filter_map` -> `map`. It only saves one line of code, and that can be attributed to different formatting of the `EnumsChild` match--in fact, the same syntax tweak would result in the original form being shorter still.

Reply:
The same comment as below applies to this... the idea of `get_variant` hadn't occurred to me when I started, so I was initially focusing on trying to unnest things and break them into small clear repeated steps. I'm thinking of redoing the commits so that I define and use `get_variant` first, then unnest in a later commit (I was thinking that first shrinking the code with `get_variant` might also making it easier to see that the unnesting doesn't change behavior).

Reply:
It hit me that maybe I should describe why splitting up the code this way appealed to me: It's easier for me to interpret XML-related iters if all intermediate steps are constrained to forms `.filter_map(get_variant!(Foo::BarVariant))` or `.flat_map(&foo.children)` and only the last step has custom logic. Doing that allows me to read the iterator definition as a path: "traverse into `BarVariant` elements, then traverse into their children, then... etc... until we've reached the elements we want to process, then we apply unique application logic to those elements"

Reply:
I'll go ahead leave this change out of the latest PR.